src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/autocaseupdate/AutoCaseUpdaterTest.java

```diff
@@ -0,0 +1,162 @@
+
+        // then
+        assertThat(result).isEqualTo(AutoCaseUpdateResult.ABANDONED);
+
+        verifyNoMoreInteractions(caseUpdateDataService);
+        verifyNoMoreInteractions(ccdApi);
+        verifyNoMoreInteractions(caseDataBuilder);
+    }
+
+    @Test
+    void should_return_error_if_ccd_operation_fails() {
+        // given
+        Envelope envelope = sampleEnvelope();
+        given(ccdApi.getCaseRefsByEnvelopeId(envelope.id, envelope.container))
+            .willThrow(RuntimeException.class);
+
+        // when
+        var result = service.updateCase(envelope);
+
+        // then
+        assertThat(result).isEqualTo(AutoCaseUpdateResult.ERROR);
+    }
+
+    @Test
+    void should_update_case_if_only_one_case_is_found() {
+        // given
+        Long existingCaseRef = 1L;
+        CaseDetails existingCaseDetails = sampleCaseDetails();
+        Envelope envelope = sampleEnvelope();
+        SuccessfulUpdateResponse updateDataResponse = sampleUpdateDataResponse();
+
+        given(ccdApi.getCaseRefsByEnvelopeId(envelope.id, envelope.container))
+            .willReturn(List.of(existingCaseRef));
+
+        given(ccdApi.getCase(existingCaseRef.toString(), envelope.jurisdiction))
+            .willReturn(existingCaseDetails);
+
+        given(caseUpdateDataService.getCaseUpdateData(envelope.container, existingCaseDetails, envelope))
+            .willReturn(updateDataResponse);
+
+        // when
+        var result = service.updateCase(envelope);
+
+        // then
+        assertThat(result).isEqualTo(AutoCaseUpdateResult.OK);
+
+        // should request data from appropriate service
+        verify(caseUpdateDataService)
+            .getCaseUpdateData(envelope.container, existingCaseDetails, envelope);
+
+        // should call CCD to update given case
+        verify(ccdApi)
+            .updateCase(
+                eq(existingCaseDetails.getJurisdiction()),
+                eq(existingCaseDetails.getCaseTypeId()),
+                eq(EventIds.ATTACH_SCANNED_DOCS_WITH_OCR),
+                eq(existingCaseRef.toString()),
+                caseBuilderCaptor.capture(),
+                anyString()
+            );
+
+        // and
+        Function<StartEventResponse, CaseDataContent> caseBuilder = caseBuilderCaptor.getValue();
+
+        var startEventResponse = mock(StartEventResponse.class);
+        given(startEventResponse.getEventId()).willReturn("event-id");
+        given(startEventResponse.getToken()).willReturn("event-token");
+
+        given(
+            caseDataBuilder
+                .getCaseDataContent(
+                    updateDataResponse.caseDetails.caseData,
+                    envelope.id,
+                    "event-id",
+                    "event-token"
+                )
+        ).willReturn(caseDataContent);
+
+        CaseDataContent caseDataContentSentToCcd = caseBuilder.apply(startEventResponse);
+
+        assertThat(caseDataContentSentToCcd).isEqualTo(caseDataContent);
```

Comment:
It's so troublesome to test this (function passed as argument to an injected service).
This is the pattern used in other places dealing with `ccdApi`.
Any other ideas?